include/boost/geometry/algorithms/detail/overlay/get_turn_info.hpp

```diff
@@ -131,6 +131,17 @@ struct base_turn_handler
             ? 1 : 0;
     }
 
+    template <typename Point1, typename Point2>
+    static inline typename geometry::coordinate_type<Point1>::type
+            distance_measure(Point1 const& a, Point2 const& b)
+    {
+        // TODO: use comparable distance for point-point instead - but that
+        // causes currently cycling include problems
+        typedef typename geometry::coordinate_type<Point1>::type ctype;
+        ctype const dx = get<0>(a) - get<0>(b);
+        ctype const dy = get<1>(a) - get<1>(b);
+        return dx * dx + dy * dy;
```

Comment:
Right, there should be similar tool as `equals_point_point` where you only include the algorithm for points. But this tool should take the strategy so it'd be ok to simply call comparable distance strategy here.

This code is cartesian-only, is that correct?

Reply:
Good point. Not really, I will harmonize it.